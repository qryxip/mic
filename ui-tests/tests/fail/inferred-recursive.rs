use mic::{answer, solve};

#[answer]
fn main() -> _ {
    if false {
        main();
    }
    ""
}

#[solve]
fn solve() -> _ {
    if false {
        solve();
    }
    ""
}
