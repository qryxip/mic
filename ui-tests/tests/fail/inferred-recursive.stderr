error[E0618]: expected function, found `__YouCannotRecurseIfTheOutputTypeIsInferred`
 --> tests/fail/inferred-recursive.rs:6:9
  |
6 |         main();
  |         ^^^^--
  |         |
  |         call expression requires function

error[E0618]: expected function, found `__YouCannotRecurseIfTheOutputTypeIsInferred`
  --> tests/fail/inferred-recursive.rs:14:9
   |
14 |         solve();
   |         ^^^^^--
   |         |
   |         call expression requires function
